workers/unity/Packages/com.improbable.gdk.mobile/Editor/MobileLauncherWindow.cs

```diff
@@ -0,0 +1,173 @@
+
+        private static readonly Color DarkGrey = new Color(0.4f, 0.4f, 0.4f, 1);
+        private static readonly Color LightGrey = new Color(0.7f, 0.7f, 0.7f, 1);
+
+        private MobileLaunchConfig launchConfig;
+
+        private string[] simulatorNames;
+        private string[] deviceNames;
+        
+        private int simulatorNameIndex;
+        private int deviceNameIndex;
+
+        [MenuItem("SpatialOS/Mobile launcher", false, 70)]
+        public static void ShowWindow()
+        {
+            var inspectorWindowType = typeof(EditorWindow).Assembly.GetType("UnityEditor.InspectorWindow");
+            var deploymentWindow = GetWindow<MobileLauncherWindow>(inspectorWindowType);
+            deploymentWindow.titleContent.text = "Mobile Launcher";
+            deploymentWindow.titleContent.tooltip = "A tab for managing your mobile client-workers.";
+            deploymentWindow.Show();
+        }
+
+        private void OnEnable()
+        {
+            launchConfig = new MobileLaunchConfig();
+            
+#if UNITY_EDITOR_OSX
+            deviceNames = iOSLaunchUtils.RetrieveAvailableiOSDevices().Keys.ToArray();
+            simulatorNames = iOSLaunchUtils.RetrieveAvailableiOSSimulators().Keys.ToArray();
+#endif
+        }
+
+        public void OnGUI()
+        {
+            GUILayout.Label(MobileSectionLabel, EditorStyles.boldLabel);
+            using (new EditorGUI.IndentLevelScope())
+            {
+                launchConfig.RuntimeIp = EditorGUILayout.TextField(RuntimeIpLabel, launchConfig.RuntimeIp);
+                launchConfig.ShouldConnectLocally = EditorGUILayout.Toggle(ConnectLocallyLabel, launchConfig.ShouldConnectLocally);
+            }
+            
+            DrawHorizontalLine(10, LightGrey);
+            GUILayout.Label(AndroidSectionLabel, EditorStyles.boldLabel);
+
+            if (GUILayout.Button("Launch Android app"))
+            {
+                AndroidLaunchUtils.Launch(launchConfig.ShouldConnectLocally, launchConfig.RuntimeIp);
+            }
+
+#if UNITY_EDITOR_OSX
+
+            DrawHorizontalLine(10, LightGrey);
+
+            GUILayout.Label(iOSSectionLabel, EditorStyles.boldLabel);
+
+            using (new EditorGUI.IndentLevelScope())
+            {
+                launchConfig.DevelopmentTeamId = EditorGUILayout.TextField(DevelopmentTeamIdLabel, launchConfig.DevelopmentTeamId);
+                
+                if (GUILayout.Button("Build XCode project"))
+                {
+                    iOSLaunchUtils.Build(launchConfig.DevelopmentTeamId);
+                }
+
+                DrawHorizontalLine(8, DarkGrey);
+
+                using (new GUILayout.HorizontalScope())
+                {
+                    using (new EditorGUI.DisabledScope(simulatorNames.Length == 0))
+                    {
+                        simulatorNameIndex =
+                            EditorGUILayout.Popup("Simulator Model", simulatorNameIndex, simulatorNames);
```

Comment:
Is this okay where the index is 0 and the names array is empty?

Reply:
it's in a disabled scope, so I would assume it's fine

Reply:
Well that doesn't stop the code from executing I don't think (it still renders the element you just can't interact with it)

Reply:
Discussed offline - works properly!